tests/convert_secret_while_to_static_for/secret_dependent_while_loop.mlir

```diff
@@ -0,0 +1,47 @@
+// RUN: heir-opt --convert-secret-while-to-static-for %s | FileCheck %s
+
+// CHECK-LABEL: @basic_while_loop_with_secret_condition
+func.func @basic_while_loop_with_secret_condition(%input: !secret.secret<i16>) -> !secret.secret<i16> {
+  // CHECK-NOT: scf.while
+  // CHECK: affine.for
```

Comment:
While I think it's fine to just test replacement of scf.while to affine.for, it might be worth checking a few properties of the result, importantly maybe that the affine.for loop's iter_arg value starts at the block arg for the secret value `%input`, and that there is an `scf.if` statement created with the result of the condition.